src/traces/box/attributes.js

```diff
@@ -255,5 +255,16 @@ module.exports = {
             'Do the hover effects highlight individual boxes ',
             'or sample points or both?'
         ].join(' ')
-    }
+    },
+    vwidth: {
+        valType: 'number',
+        min: 0,
+        role: 'info',
+        dflt: false,
```

Comment:
(also this is in `box/attributes` but the `description` below says violins...)

Reply:
I was thinking of [`nticks`](https://github.com/plotly/plotly.js/blob/7ae6fd68533d6489e3161cdbdfe2a44217f530e1/src/plots/cartesian/layout_attributes.js#L224), [`nbins(x|y)`](https://github.com/plotly/plotly.js/blob/7ae6fd68533d6489e3161cdbdfe2a44217f530e1/src/traces/histogram/attributes.js#L132), also looks like we do that for [`maxdisplayed`](https://github.com/plotly/plotly.js/blob/7ae6fd68533d6489e3161cdbdfe2a44217f530e1/src/traces/scatter/attributes.js#L360).

`null` is certainly more meaningful on its own than `0`, and it's necessary if `0` is a valid value distinct from "auto" - like [`bar.offset`](https://github.com/plotly/plotly.js/blob/7ae6fd68533d6489e3161cdbdfe2a44217f530e1/src/traces/bar/attributes.js#L142) where `0` means "center the bar at the data value" and "auto" means "if the bars are grouped, offset them all so they end up side-by-side".

It's just a little funny though with `valType: 'number'` - it means `null` looks invalid, so if you try and set it we'll determine it to be invalid and fall back on the default, which I guess is fine as long as `null` *is* the default, but it would still fail `Plotly.validate`. Also you can't set it at all via `restyle/relayout`, because `null` there means "unset" - again, not immediately a problem as long as this is the default. BUT... what if you have a template that overrides the default? Seems to me then you would be unable to revert to auto with any setting at all of the user attribute, right?

`valType: 'angle'` adds a special value `'auto'`, which might be the best of both worlds? Its meaning is clear (and distinct from `0`), and it's not intercepted by plotlyjs like `null` is. Perhaps we should allow `valType: 'number'` to specify `allowAuto: 'true'`, or `extras: ['auto']`?

Reply:
> BUT... what if you have a template that overrides the default? Seems to me then you would be unable to revert to auto with any setting at all of the user attribute, right?

Interesting point. This is somewhat related to https://github.com/plotly/plotly.js/issues/2834, I'll comment over there.